### ISSUE TITLE
Adds incorrect --add-opens with JRuby 9.1

### ISSUE THREAD

**Issue by mvz** [jruby/jruby-launcher#28]
Version 1.1.10 of this gem includes code to add `--add-opens` options. However, the module specified, `org.jruby.dist`, does not exist in JRuby 9.1, causing warnings to be emitted.

This [seems](https://travis-ci.community/t/incorrect-add-opens-set-for-jruby-9-1-17-on-xenial/6331) to be the cause of [build failures](https://travis-ci.org/troessner/reek/builds/611542039) that I'm seeing on Travis CI.

**Comment by headius** [comment]
Ah yes, I suspected this was the cause. We will need to tweak the gem to not add those flags under versions that do not have the module information.

May I ask why you can't update to 9.2.9?

**Comment by mvz** [comment]
We also build on latest JRuby 9.2, but officially support 9.1 as well. This test build only built on 9.1 to avoid running extra builds on Travis while debugging the failure.

**Comment by headius** [comment]
Ok thank you. I'll see about getting the launcher fixed and released this week.

**Comment by headius** [comment]
Just clarifying for anyone following this issue: these are indeed just **warnings**, and they normally don't interfere with JRuby operation (though they are indeed rather annoying).

**Comment by headius** [comment]
Reproduced with JRuby 9.1.17.0 and latest jruby-launcher:

```
$ jruby -v -e 1
WARNING: Unknown module: org.jruby.dist specified to --add-opens
WARNING: Unknown module: org.jruby.dist specified to --add-opens
WARNING: Unknown module: org.jruby.dist specified to --add-opens
jruby 9.1.17.0 (2.3.3) 2018-04-20 d8b1ff9 OpenJDK 64-Bit Server VM 11.0.4+11 on 11.0.4+11 +jit [darwin-x86_64]
```

**Comment by halostatue** [comment]
@headius …unfortunately for me, this breaks some tests for diff-lcs which check logging. See https://travis-ci.org/halostatue/diff-lcs/jobs/633999940 for an example. Is there a way to turn this warning off in Travis, as an example?

**Comment by headius** [comment]
@halostatue Unfortunately not. The OpenJDK folks did not provide a way to completely disable the warning. It will always show at least once without proper `add-opens` flags.

**Comment by headius** [comment]
@halostatue Are you running a current JRuby or JRuby 9.1.x? This issue only affects 9.1.x, and 9.2 should work fine with the bash script it ships.

**Comment by halostatue** [comment]
I’m not actually running JRuby; I’m running JRuby on Travis for validation of diff-lcs. The test in question fails because I’m not expecting any output in stderr. It _only_ fails on 9.1.x, and I don’t really want to mark 9.1.x as “allowed to fail” in Travis or write some conditional code in the test for JRuby 9.1.

**Comment by headius** [comment]
JRuby 9.1 is basically EOL at this point. You'd be fine excluding it.

**Comment by headius** [comment]
I see your failures...hmmm. I will see if I can get a new version of the launcher out that at least does not supply these values on JRuby 9.1.

**Comment by halostatue** [comment]
If JRuby 9.1 is actually EOL, I have no problem soft-deprecating it (removing it from Travis or allowing failures).

**Comment by headius** [comment]
This should have been fixed as of the 1.1.13 release, which does not add module flags at all for JRuby versions 9.2.0.0 or earlier.
